cmd/cstor-pool-mgmt/volumereplica/volumereplica.go

```diff
@@ -699,24 +700,127 @@ func GetAndUpdateSnapshotInfo(
 	}
 
 	// If CVR is in rebuilding go and get snapshot information
-	// from other replicas
-	//if cvr.Status.Phase == apis.CVRStatusRebuilding {
-	//	err = getAndUpdatePendingSnapshotsList(clientset, cvr)
-	//}
+	// from other replicas and add snapshots under pending snapshot list
+	if cvr.Status.Phase == apis.CVRStatusRebuilding ||
```

Comment:
Should we update for status!=healthy? replica controller have sync period of 30s and that may delay the status update further.

Reply:
Estimating rebuild on the replica which is undergoing rebuild is good and we shouldn't bother about other replicas. 

IMO we shouldn't go get the snapshot info from peer CVR if current CVR is not in Healthy  state(Because for every 30 sec n volumes making etcd call is not good).

Reply:
this can be called only when there is status.Phase transition from Degraded to Rebuilding or ReconstructingNewReplica to avoid unnecessary calls?

Reply:
ok.. this need to be called in every sync for Rebuilding or ReconstructingNewReplica state for the case where snapshots are created/deleted during the rebuild of this CVR in other replicas..
we need to cover this case in our testing be it either BDD or E2E

Reply:
Yes that is the case